core/src/main/java/software/amazon/awssdk/http/pipeline/stages/HandleResponseStage.java

```diff
@@ -152,16 +150,4 @@ private void closeInputStreamIfNeeded(HttpResponse httpResponse,
         }
     }
 
-    /**
-     * Determines whether a request failed or not based on the response. If the response is an empty optional then
-     * execution failed with some non-service error like an IOException or some other client error. If the response
-     * is a fulfilled optional and {@link Response#isFailure()} is true then the execution failed with a service error
-     * of some kind.
-     *
-     * @param response Optional of unmarshalled response.
-     * @return True if the response was a failure. False if it was a success.
-     */
-    private boolean didRequestFail(Response<OutputT> response) {
```

Comment:
Lol nice. I'd like to think at one point it did more than this.

Reply:
It did!